onnxruntime/core/providers/coreml/builders/impl/reshape_op_builder.cc

```diff
@@ -40,7 +40,7 @@ Status ReshapeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
                                                const logging::Logger& logger) const {
   const auto& input_defs = node.InputDefs();
   std::vector<int64_t> input_shape;
-  ORT_RETURN_IF_NOT(GetStaticShape(*input_defs[0], input_shape, logger), "Cannot get shape of data");
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape of data");
```

Comment:
hm, it looks like ReshapeHelper does not like an input shape with dynamic dimensions:
https://github.com/microsoft/onnxruntime/blob/30415e6a3660a4bcbcc9d39ba317686faa628ed4/onnxruntime/core/providers/cpu/tensor/reshape_helper.h#L15-L16

so even if GetShape were to succeed here, if input_shape has a -1 dimension, ReshapeHelper will complain.